### PR TITLE
Ensure BED files are tab-delimited, sorted, and include headers 

### PR DESCRIPTION
**Pull request by kwentine**
- Ensure BED files are tab-delimited
  - As [discussed](https://scilifelab.atlassian.net/browse/EB-131?focusedCommentId=15732) indiscriminately replace every sequence of spaces by a single tab. Let's see how far that gets us.
- Sort BED files while preserving the header lines

### CODE REVIEW

File: scripts/filter [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+    rm -f "${tmp}"
+}
+
+# Replace spaces by tabs
+# Output:
+#  Input stream with every whitespace sequence replaced by a
+#  single tab character.
+tabify() {
+    sed -E 's/ +/\t/g'
+}
```
Comment (RMCrean) [inline scripts/filter:24]: Not sure if this is an edge case that could actually happen or not but if there was both a space and tab then then the original tab would also be kept. Could perhaps be changed to:
`sed -E 's/\s+/\t/g'`


As an example, this input:
```
input="chr1  100  200\n\
chr1  150\t 250\n\
chr2\t100 200"
```

would give: 
```
Original tabify function output:
chr1    100     200
chr1    150             250
chr2    100     200

Changed tabify function output:
chr1    100     200
chr1    150     250
chr2    100     200
```

**Review by RMCrean — APPROVED** [review]
I'm not sure how relevant/realistic my comment is. LGTM :)

**Review by brinkdp — APPROVED** [review]
Thanks, looks good! Tested with `dockermake` and a few different BED files (with and without headers; space or tab delimited). All resulted in working JBrowse instances.

(I love the branch name, by the way!)

File: scripts/filter (hunk 1 above)
Comment (brinkdp) [reply scripts/filter:24]: Good thinking. I would however consider this an edge-case 😄. A handcrafted BED could indeed end up having this typo, but most BEDs we will encounter have been produced by established bioinformatics tools that (I assume) will apply consistent formatting.

Comment (kwentine) [reply scripts/filter]: Thanks for the suggestion, I think it is a generalization that will safely constrain the output. Added in 47e1911

**Pull request merged**